src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -68,7 +68,7 @@ public async Task<UpsertOutcome> UpsertAsync(ResourceWrapper resource, WeakETag
             int etag = 0;
             if (weakETag != null && !int.TryParse(weakETag.VersionId, out etag))
             {
-                throw new ResourceConflictException(weakETag);
+                etag = -1;
```

Comment:
This would be a good place to add a comment explaining why :)

Reply:
Thought you might say that... added in f9a779a!